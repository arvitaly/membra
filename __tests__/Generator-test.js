"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const schema2_1 = require("./../__fixtures__/schema2");
const Generator_1 = require("./../Generator");
fdescribe("Generator", () => {
    it("query", () => {
        const generator = new Generator_1.default(schema2_1.default);
        const execution = generator.generate((schemaObj) => {
            const items = schemaObj.query.viewer.model1({
                first: 10,
                after: "15",
                before: "15",
                last: 10,
            }).edges.map(({ node }) => {
                const model2 = node.model2({
                    where: { field1: "test" },
                });
                return {
                    f: node.field1,
                    x: model2.field2.toFixed(),
                    y: model2.id.toLowerCase(),
                };
            });
            return {
                items,
                id: schemaObj.query.viewer.id,
            };
        });
        const query = generator.getQuery(execution);
        expect(query).toMatchSnapshot();
        const result = execution.schemaObj.fillData({
            viewer: {
                id: "15",
                model1: {
                    edges: [{
                            node: {
                                field1: "field1Value",
                                model2: {
                                    field2: 100.888,
                                    id: "11",
                                },
                            },
                        }],
                },
            },
        }, execution.executor);
        expect(result).toMatchSnapshot();
    });
});
