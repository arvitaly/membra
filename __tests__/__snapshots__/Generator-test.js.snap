// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator mutation 1`] = `
"mutation {
    createModel1(input: {setField1: {field1: \\"11\\"}}) {
        test
    }
}"
`;

exports[`Generator mutation 2`] = `"StopHi"`;

exports[`Generator query 1`] = `
"query {
    viewer {
        model1(first: 10, after: \\"15\\", before: \\"15\\", last: 10) {
            edges {
                node {
                    model2(where: {field1: [\\"test\\"], field3: [{field4: \\"testValueField4\\"}]}) {
                        field2
                        id
                    }
                    field1
                }
            }
        }
        id
    }
}"
`;

exports[`Generator query 2`] = `
Object {
  "id": "15",
  "items": Array [
    Object {
      "f": "field1Valuevar1Value2",
      "x": "101",
      "y": "11",
    },
  ],
}
`;

exports[`Generator query with same arguments 1`] = `
"query {
    viewer {
        model1(first: 10) {
            edges {
                node {
                    field1
                }
            }
        }
        f1:model1(first: 20) {
            edges {
                node {
                    field1
                }
            }
        }
    }
}"
`;
